ext/msgpack/factory_class.c

```diff
@@ -208,7 +209,7 @@ static VALUE Factory_register_type(int argc, VALUE* argv, VALUE self)
         }
     }
 
-    msgpack_unpacker_ext_registry_put(&fc->ukrg, ext_module, ext_type, unpacker_proc, unpacker_arg);
+    fc->ukrg = msgpack_unpacker_ext_registry_put(fc->ukrg, ext_module, ext_type, unpacker_proc, unpacker_arg);
```

Comment:
On this change, `msgpack_unpacker_ext_registry_put` doesn't put an entry into `fc->ukrg` but actually copy and add and entry then return it.
So I think it's better to replace `fc->ukrg` with the newer one directly. Or to rename it.